tck/faces23/ajax/src/test/java/ee/jakarta/tck/faces/test/javaee8/ajax_selenium/Spec1423IT.java

```diff
@@ -0,0 +1,123 @@
+        WebPage page = getPage("spec1423.xhtml");
+        WebElement button;
+
+        assertTrue(page.findElement(By.id("scriptResult")).getText().isEmpty());
+        assertTrue(page.findElement(By.id("stylesheetResult")).getText().isEmpty());
+
+        button = page.findElement(By.id("form1:addViaHead"));
+        button.click();
+        page.waitReqJs(Duration.ofMillis(6000));
+        assertTrue(page.findElement(By.id("scriptResult")).getText().trim().equals("addedViaHead"));
+        assertTrue(page.findElement(By.id("stylesheetResult")).getText().isEmpty());
+    }
+
+    @Test
+    public void testSpec1423() throws Exception {
+        WebPage page = getPage("spec1423.xhtml");
+        WebElement button;
+
+        assertTrue(page.findElement(By.id("scriptResult")).getText().isEmpty());
+        assertTrue(page.findElement(By.id("stylesheetResult")).getText().isEmpty());
+
+        button = page.findElement(By.id("form1:addViaHead"));
+        button.click();
+
+        page.waitForCondition(wd -> {
+            return page.findElement(By.id("stylesheetResult")).getText().trim().isEmpty() &&
+                    page.findElement(By.id("scriptResult")).getText().trim().equals("addedViaHead");
+        });
+
+
+        assertTrue(page.findElement(By.id("scriptResult")).getText().trim().equals("addedViaHead"));
+        assertTrue(page.findElement(By.id("stylesheetResult")).getText().isEmpty());
+
+        button = page.findElement(By.id("form2:addViaInclude"));
+        button.click();
+
+        page.waitForCondition(wd -> {
+            return page.findElement(By.id("stylesheetResult")).getText().trim().equals("rgb(255, 0, 0)") &&
+                    page.findElement(By.id("scriptResult")).getText().trim().equals("addedViaInclude");
+        });
+
+        button = page.findElement(By.id("form1:addViaBody"));
+        button.click();
+        page.waitReqJs(Duration.ofMillis(6000));
+        assertTrue(page.findElement(By.id("scriptResult")).getText().trim().equals("addedViaBody"));
+        assertTrue(page.findElement(By.id("stylesheetResult")).getText().trim().equals("rgb(255, 0, 0)"));
+
+        button = page.findElement(By.id("form2:addViaInclude"));
+        button.click();
+
+        page.waitReqJs(Duration.ofMillis(6000));
+        assertTrue(page.findElement(By.id("scriptResult")).getText().trim().equals("addedViaBody"));
+        assertTrue(page.findElement(By.id("stylesheetResult")).getText().trim().equals("rgb(255, 0, 0)"));
+
+        button = page.findElement(By.id("form1:addProgrammatically"));
+        button.click();
+        // wait for condition implictly also handles the
+
+        page.waitForCondition(wd -> {
+            return page.findElement(By.id("stylesheetResult")).getText().trim().equals("rgb(0, 255, 0)") &&
+                    page.findElement(By.id("scriptResult")).getText().trim().equals("addedProgrammatically");
+        });
+
+
+        button = page.findElement(By.id("form1:addViaHead"));
```

Comment:
It's probably better if page was reloaded so that the script/style results of previous test step are guaranteed to be cleared out.